src/clj_kondo/impl/analyzer.clj

```diff
@@ -1276,34 +1276,53 @@
 
 (defn analyze-protocol-impls [ctx defined-by ns-name children]
   (loop [current-protocol nil
-         protocol-ns nil
-         protocol-name nil
          children children]
     (when-first [c children]
       (if-let [sym (utils/symbol-from-token c)]
-        ;; we have encountered a protocol or interface name
-        (do (analyze-expression** ctx c)
-            (let [{protocol-ns :ns protocol-name :name} (resolve-name ctx ns-name sym)]
-              (recur sym protocol-ns protocol-name (rest children))))
+        ;; We have encountered a protocol or interface name, or a
+        ;; record or type name (in the case of extend-protocol and
+        ;; extend-type). We need to deal with extend-protocol in a
+        ;; special way, as there is a single protocol being extented
+        ;; to multiple records/types.
+        (do
+          (analyze-expression** ctx c)
+          (recur (case (name defined-by)
```

Comment:
You can dispatch on symbols in case, but be aware that you do not need to quote them:

```
(case defined-by
  extend-protocol ...)
```

Reply:
I think I get why you used name: to strip the clojure.core namespace?